### PR TITLE
Pin Rust toolchain, use Rustls

### PR DESCRIPTION
**Pull request by calyptobai**
Based on our `flake.nix` file, it seems we have the following runtime dependencies: `openssl.out rocksdb git zlib nsync onnxruntime14`. Judging from our own `Cargo.lock`, and examining qdrant's `Cargo.lock`, we know that:

- [`rocksdb` is built statically and not needed at runtime](https://github.com/rust-rocksdb/rust-rocksdb/blob/master/librocksdb-sys/Cargo.toml#L16)  (unless provided by environment variables, which we don't do in default builds)
- [`zlib` is not used as default features are deactivated](https://github.com/qdrant/qdrant/blob/master/lib/segment/Cargo.toml#L28)
- [We removed `libgit2` in favor of `gitoxide`](https://github.com/BloopAI/bloop/pull/432)
- [We pin `onnxruntime` versions to specific binaries provided by `ort`](https://github.com/BloopAI/bloop/pull/231)

We also add a Rust toolchain configuration to pin the toolchain at `1.70.0`, and ensure that `openssl-sys` is built statically using the `vendored` feature.

This removes all remaining non-platform-specific runtime dependencies except `nsync`, which does not seem to be linked into either `bloop` builds or `qdrant`'s builds.

### CODE REVIEW

File: server/bleep/Cargo.toml [hunk 1]
```diff
@@ -110,6 +110,9 @@ jsonwebtoken = { version = "8.3.0", features = ["use_pem"] }
 tiktoken-rs = "0.4.2"
 semver = { version = "1", features = ["serde"] }
 
+# vendored deps
+openssl-sys = { version = "0.9.88", features = ["vendored"] }
```
Comment (rsdy) [inline server/bleep/Cargo.toml:114]: We don't currently even transitively depend on `openssl`, so I don't see a reason why we would bring this dependency in.

I also think `vendored` openssl likely won't work for us, as it doesn't use the system cert store, so we'll need additional work to make HTTPS through it functional.

**Review by rsdy — CHANGES_REQUESTED** [review]

File: server/bleep/Cargo.toml (hunk 1 above)
Comment (calyptobai) [reply server/bleep/Cargo.toml:114]: > We don't currently even transitively depend on `openssl`, so I don't see a reason why we would bring this dependency in.

What do you mean?

```
openssl-sys v0.9.88
├── native-tls v0.2.11
│   ├── hyper-tls v0.5.0
│   │   └── reqwest v0.11.18
│   │       ├── bleep v0.4.5
...
```

> I also think `vendored` openssl likely won't work for us, as it doesn't use the system cert store, so we'll need additional work to make HTTPS through it functional.

Do you have an example of this failing? I've tested on linux and Windows, and it doesn't seem to have any issues. We can use `openssl-probe` if there are broken examples.

Comment (rsdy) [reply server/bleep/Cargo.toml:114]: Oh, in this case a better fix would be to just use `rustls` for `rustls-tls-native-roots` feature on `reqwest` and avoid openssl altogether.

Comment (rsdy) [reply server/bleep/Cargo.toml:114]: It seems like there are a few things going on here.
 1. OpenSSL is occasionally pulled in through `hyper-tls`, depending on host platform
 2. `bleep` pulled in a few changes recently that subtly add a dependency here, but these are easy to fix and switch to `rustls` based on `cargo tree` output.
 3. `tauri` currently only supports vendored and unvendored OpenSSL, and our app config uses the vendored version, so the library (I have to assume) will be correctly initialized by Tauri. However, their `dev` branch already supports rustls, so I have to assume sooner or later they'll release it.
 
 I think the meaningful way forward here is to make sure `bleep` builds without `hyper-tls`, and deal with Tauri when they release their new feature configuration.

Comment (calyptobai) [reply server/bleep/Cargo.toml:114]: As discussed offline, we can tackle this in another PR, given it seems certificates load correctly on all platforms

**Review by rsdy — APPROVED** [review]

**Pull request merged**